### ISSUE TITLE
Accidentally disconnected from Server

### ISSUE THREAD

**Issue by kanishkaganguly** [stream-pi/server#33]
Back again with another issue/bug.

```bash
INFO: action ID : 189cc7de-5666-420d-9039-c2736271e8a0
Invalid : false
COLOr :
COLOr :
Mar 13, 2021 7:03:39 PM com.stream_pi.server.window.dashboard.actiongridpane.ActionGridPane renderActions
INFO: action ID : c3d8467b-f5b5-48ba-a755-aa057a4516a7
Invalid : false
COLOr :
COLOr :
new alphas with length = 12288
ES2ResourceFactory: Prism - createStockShader: Texture_ImagePattern.frag
Mar 13, 2021 7:03:43 PM com.stream_pi.server.connection.ClientConnection run
SEVERE: null
java.io.EOFException
        at java.base/java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:3171)
        at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1630)
        at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:493)
        at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:451)
        at com.stream_pi.server@1.0.0-SNAPSHOT/com.stream_pi.server.connection.ClientConnection.run(ClientConnection.java:197)
true 22222222222222222222222222222222222222222
com.stream_pi.util.exception.MinorException: Accidentally disconnected from GangulyStreamClient.
        at com.stream_pi.server@1.0.0-SNAPSHOT/com.stream_pi.server.connection.ClientConnection.run(ClientConnection.java:244)
Mar 13, 2021 7:03:43 PM com.stream_pi.server.controller.Controller handleMinorException
SEVERE: Accidentally disconnected from GangulyStreamClient.
com.stream_pi.util.exception.MinorException: Accidentally disconnected from GangulyStreamClient.
        at com.stream_pi.server@1.0.0-SNAPSHOT/com.stream_pi.server.connection.ClientConnection.run(ClientConnection.java:244)

com.stream_pi.util.exception.MinorException: Accidentally disconnected from GangulyStreamClient.
        at com.stream_pi.server@1.0.0-SNAPSHOT/com.stream_pi.server.connection.ClientConnection.run(ClientConnection.java:244)
```
Randomly disconnects, that I think might be related to setting icon backgrounds. Doesn't stay connected longer than a few seconds.

**Comment by rnayabed** [comment]
Which server and client platform?

**Comment by kanishkaganguly** [comment]
> Which server and client platform?

I am on Android (client) and Ubuntu 18.04 (server)
I am running the latest 1.0.0 EA2 release from the repository.

**Comment by rnayabed** [comment]
This looks like a problem due to the fact that android sleeps after sometime. Can you try with "caffiene" mode ON on android and see if the problem persists?

**Comment by kanishkaganguly** [comment]
I have kept my phone on "Always On when on charge" from Developer Options. The screen remains on, and the app remains in focus. Is that the same as what Caffeine mode would do?

**Comment by rnayabed** [comment]
Are you using Gifs?

**Comment by kanishkaganguly** [comment]
No, only jpg and png.

**Comment by rnayabed** [comment]
seems like outofmemory. can you get me logcat from your device?

**Comment by rnayabed** [comment]
does the android app crash too?

**Comment by kanishkaganguly** [comment]
No, neither the server or the client "crashes". I just have to reconnect again, for things to resume working.
Let me try for a logcat, any particular way you would like me to record it?

**Comment by rnayabed** [comment]
Just record everything

**Comment by kanishkaganguly** [comment]
Here are the server and client logs.
[client.txt](https://github.com/stream-pi/server/files/6152097/client.txt)
[server.txt](https://github.com/stream-pi/server/files/6152098/server.txt)

**Comment by rnayabed** [comment]
The Server says that the pipe breaks in between. Unusual.

However, the client logcat is incomplete. Best would be to start logging logcat before running the app. Run the app, while logging logcat on PC, then stop when the app finally crashes.

**Comment by kanishkaganguly** [comment]
The app never crashes, actually. It just disconnects gracefully. I used logcat to only report data from a specific app, so whatever you see in the file happens only after I start the client app.

**Comment by rnayabed** [comment]
Can you record logcat using adb on PC?

**Comment by kanishkaganguly** [comment]
I did, this was recorded from my PC through Android Studio and adb.

**Comment by rnayabed** [comment]
The logcat is incomplete. There is no mention of init for JavaFX. Its probably started recording after the app started.

**Comment by rnayabed** [comment]
Can you record it using `adb logcat` directly from terminal  @kanishkaganguly ?

**Comment by rnayabed** [comment]
Closing this since inactive.

**Comment by rnayabed** [comment]
Re open if needed

**Comment by kanishkaganguly** [comment]
Sorry for the inactivity, I was AFK for several days. Here are the latest logs as requested, captured directly as follows:

`./run 2>&1 | tee logs/server_logs.txt`
`adb logcat 2>&1 | tee logs/client_logs.txt`

Same error as before, as soon as I try to set the icon and then select **Media Key --> Volume Down** it disconnects from server.
[client_logs.txt](https://github.com/stream-pi/server/files/6253402/client_logs.txt)
[server_logs.txt](https://github.com/stream-pi/server/files/6253404/server_logs.txt)

**Comment by rnayabed** [comment]
Can you try making the "battery optimisation" in android settings, for Stream-Pi Client to 'Do not Optimize' and see if this behavior happens again?

Can you also give the model and android version of your phone?

**Comment by kanishkaganguly** [comment]
Samsung S7 Edge: SM-G935F
Android version: 7.0

Even when the "battery optimization" is disabled, I get the same error. Doesn't that kick in only when the app is not in use?
Also, I think I had mentioned in an earlier post, I have screen set to "Always On" from the Developer Options menu.

**Comment by rnayabed** [comment]
A lot has been changed, try reproducing the error by using one of the snapshot builds (use snapshot for both server and client)

**Comment by rnayabed** [comment]
Reopen if issue persists with 1.0.0-EA+3-snapshot or new
